packages/core/src/table/table.scss

```diff
@@ -0,0 +1,205 @@
+    border-bottom: $cds-alias-object-border-width-100 solid $cds-alias-object-border-color-tint;
+  }
+
+  tr th:first-child,
+  tr td:first-child {
+    border-left: $cds-alias-object-border-width-100 solid $cds-alias-object-border-color;
+  }
+
+  tr th:last-child,
+  tr td:last-child {
+    border-right: $cds-alias-object-border-width-100 solid $cds-alias-object-border-color;
+  }
+
+  tr:last-child td {
+    border-bottom-color: $cds-alias-object-border-color;
+  }
+
+  tfoot th {
+    border-top-width: 0;
+  }
+}
+
+// Border radius styles
+[cds-table] {
+  thead tr:first-child {
+    :first-child {
+      border-radius: 0;
+      border-top-left-radius: $cds-alias-object-border-radius-100;
+    }
+
+    :last-child {
+      border-radius: 0;
+      border-top-right-radius: $cds-alias-object-border-radius-100;
+    }
+  }
+
+  & > *:last-child > tr:last-of-type {
+    :first-child {
+      border-radius: 0;
+      border-bottom-left-radius: $cds-alias-object-border-radius-100;
+    }
+
+    :last-child {
+      border-radius: 0;
+      border-bottom-right-radius: $cds-alias-object-border-radius-100;
+    }
+  }
+}
+
+// Left Alignment options
+[cds-table~='left'] th,
+[cds-table~='left'] td,
+[cds-table] td[cds-col~='left'],
+[cds-table] th[cds-col~='left'] {
+  text-align: left;
+}
+
+[cds-table~='left'] tr th:first-child,
+[cds-table~='left'] tr td:first-child,
+[cds-table] tr td[cds-col~='left']:first-child,
+[cds-table] tr th[cds-col~='left']:first-child {
+  padding-left: $cds-global-layout-space-sm;
+}
+
+// Right Alignment options
+[cds-table~='right'] th,
+[cds-table~='right'] td,
+[cds-table] td[cds-col~='right'],
+[cds-table] th[cds-col~='right'] {
+  text-align: right;
+}
+
+// No Border options
+[cds-table~='noborder'],
+[cds-table~='noborder'] tr th,
+[cds-table~='noborder'] tr td {
+  border-width: 0 !important;
+  background-color: transparent;
+}
+
+[cds-table~='noborder'] tr th {
+  border-bottom-width: $cds-alias-object-border-width-100 !important;
+}
+
+// Compact options
+[cds-table~='compact'] {
+  td,
+  th {
+    line-height: $cds-global-typography-font-size-4;
+    padding: $cds-global-typography-font-size-1 $cds-global-typography-font-size-2;
+    padding-top: $cds-global-layout-space-sm;
+    padding-bottom: $cds-global-layout-space-xs;
+  }
+}
+
+// Vertical options
+[cds-table~='vertical'] {
+  tr td:first-child,
+  tr th:first-child {
+    background-color: $cds-alias-status-neutral-tint;
+    // color: ??;
+    border-right-width: $cds-alias-object-border-width-100;
+    border-right-style: solid;
+    border-right-color: $cds-alias-object-border-color;
+    border-bottom-color: $cds-alias-object-border-color;
+  }
+}
+
+// Highlight row
+[cds-row~='highlight'] {
+  td,
+  th {
+    background-color: $cds-alias-object-container-background-tint;
+  }
+}
+
+// Padding options
+[cds-table~='p:xxs'] {
+  th,
+  td {
+    padding: $cds-global-layout-space-xxs;
+  }
+}
+
+[cds-table~='p:xs'] {
```

Comment:
Do we want to allow for different horizontal and vertical padding as well? For layouts, this would look like `p-x:md` and `p-y:md`. I don't think we would need to go any more granular than that.

Also, what about allowing for spacing to adjust responsively? Like the tokens can do. This looks like `p@md:lg` and `p-x@sm:xs`...

Reply:
I wasn't sure this was even desirable to support or not. I did it just to see, but there could be a lot of permutations here if we want to support the layout token styles.

Reply:
If I do this, ought I make a mixin to generate these many permutations?

Reply:
I don't think the table should be doing that level of work. That should be controlled with layouts or the themes rather than a custom component specific feature. Tables cells can use layouts directly. Tables are usually dynamic so the repetitiveness is not as much of a concern.

```html
<td cds-layout="p:lg">

</td>
```

Or the padding can be controlled via themes/custom properties

```scss
[cds-table] {
  --cell-padding: $cds-global-typography-font-size-1 $cds-global-typography-font-size-2;
}

td, th {
  padding: var(--padding);
}
```

This would also better align using a compact theme rather than component specific APIs. See clubhouse story 602. This would help create compact options for datagrid and treeview as a single API solution for the entire system.

```html
<table cds-table cds-theme="compact-space">

</table>
```